internal/api/api_test.go

```diff
@@ -0,0 +1,164 @@
+package api
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/fabric8-services/fabric8-jenkins-idler/internal/cluster"
+	"github.com/fabric8-services/fabric8-jenkins-idler/internal/model"
+	"github.com/julienschmidt/httprouter"
+	"github.com/stretchr/testify/assert"
+)
+
+type JSError struct {
+	Error string
+}
+
+type mockOSC struct {
+	idleError string
+}
+
+func (m *mockOSC) Idle(apiURL string, bearerToken string, namespace string, service string) (e error) {
+	if m.idleError != "" {
+		return fmt.Errorf(m.idleError)
+	}
+	return
+}
+
+func (m *mockOSC) UnIdle(apiURL string, bearerToken string, namespace string, service string) (e error) {
+	if m.idleError != "" {
+		return fmt.Errorf(m.idleError)
+	}
+	return
+}
+
+func (m *mockOSC) IsIdle(apiURL string, bearerToken string, namespace string, service string) (int, error) {
+	if m.idleError != "" {
+		return -1, fmt.Errorf(m.idleError)
+	}
+	return 1, nil
+}
+
+func (m *mockOSC) WhoAmI(apiURL string, bearerToken string) (string, error) {
+	return "", nil
+}
+
+func (m *mockOSC) WatchBuilds(apiURL string, bearerToken string, buildType string, callback func(model.Object) error) error {
+	return nil
+}
+
+func (m *mockOSC) WatchDeploymentConfigs(apiURL string, bearerToken string, namespaceSuffix string, callback func(model.DCObject) error) error {
+	return nil
+}
+
+type mockClusterView struct{}
+
+func (m *mockClusterView) GetClusters() (r []cluster.Cluster) {
+	return
+}
+
+func (m *mockClusterView) GetDNSView() (r []cluster.DNSView) {
+	return
+}
+
+func (m *mockClusterView) GetToken(openShiftAPIURL string) (string, bool) {
+	return "", true
+}
+
+func (m *mockClusterView) String() string {
+	return ""
+}
+
+type mockResponseWriter struct {
+	buffer       bytes.Buffer
+	writerStatus int
+}
+
+func (m *mockResponseWriter) Header() (h http.Header) {
+	return http.Header{}
+}
+
+func (m *mockResponseWriter) Write(p []byte) (n int, err error) {
+	m.buffer.Write(p)
+	return len(p), nil
+}
+
+func (m *mockResponseWriter) WriteString(s string) (n int, err error) {
+	m.buffer.WriteString(s)
+	return len(s), nil
+}
+
+func (m *mockResponseWriter) WriteHeader(i int) {
+	m.writerStatus = i
+}
+
+func (m *mockResponseWriter) GetBody() string {
+	return m.buffer.String()
+}
+
+type ReqFuncType func(w http.ResponseWriter, r *http.Request, ps httprouter.Params)
+
+func Test_success(t *testing.T) {
+	mockidle := idler{
+		openShiftClient: &mockOSC{},
+		clusterView:     &mockClusterView{},
+	}
+	functions := []ReqFuncType{mockidle.Idle, mockidle.UnIdle, mockidle.IsIdle}
+
+	for _, function := range functions {
+		r, _ := http.NewRequest("GET", "/", nil)
```

Comment:
Shouldn't we use more descriptive names? `r` -> `request` ? Same goes for other variable names as well.

Reply:
I don't really care personally, r is quite standard variable name for request in go world